core/templates/dev/head/pages/exploration-editor-page/services/autosave-info-modals.service.ts

```diff
@@ -73,25 +73,36 @@ angular.module('oppia').factory('AutosaveInfoModalsService', [
             'save-version-mismatch-modal.template.html'),
           // Prevent modal from closing when the user clicks outside it.
           backdrop: 'static',
-          controller: ['$scope', function($scope) {
-            // When the user clicks on discard changes button, signal backend
-            // to discard the draft and reload the page thereafter.
-            $scope.discardChanges = function() {
-              ExplorationDataService.discardDraft(function() {
-                _refreshPage(20);
-              });
-            };
+          controller: [
+            '$scope', '$uibModalInstance',
+            function($scope, $uibModalInstance) {
+              // When the user clicks on discard changes button, signal
+              // backend to discard the draft and reload the page thereafter.
+              $scope.discardChanges = function() {
+                ExplorationDataService.discardDraft(function() {
+                  _refreshPage(20);
+                });
+              };
 
-            $scope.hasLostChanges = (lostChanges && lostChanges.length > 0);
-            if ($scope.hasLostChanges) {
-              // TODO(sll): This should also include changes to exploration
-              // properties (such as the exploration title, category, etc.).
-              $scope.lostChangesHtml = (
-                ChangesInHumanReadableFormService.makeHumanReadable(
-                  lostChanges).html());
-              $log.error('Lost changes: ' + JSON.stringify(lostChanges));
-            }
-          }],
+              // When the user clicks on save changes button, fetch the
+              // latest changes from backend.
+              $scope.saveChanges = function() {
```

Comment:
@cuichenli This button will fetch the latest data but this won't add the new changes in the change list, right? If yes, then:
- will it update all other exploration data related services with the new data?
- Why it's named "save changes" although it's not saving the changes?

Reply:
@DubeySandeep Sorry for the late reply, these days were busy on something else...

For the first question, I assume it updates all other services? Since it is fetching all related data from the backend. Sorry I am not sure about it.

For the second one, yes this function should save the data. I have updated it in a new commit(by changing ExplorationDataService.fetchLatestSavedData). Please kindly have a look. Thanks!

Reply:
> For the first question, I assume it updates all other services?

I think it won't but I'm not sure can you please check?

> .fetchLatestSavedData

This method will only fetch the latest data i.e, it won't include the latest changes you made, right? Can you please check?

Reply:
Sorry, can you elaborate about which services specifically? I will further check. Thanks!

Reply:
As for the `fetchLatestSavedData`, it will save the latest changes. 

https://github.com/oppia/oppia/blob/393a81474ddc02b7141eed80f33c720cde447517/core/templates/dev/head/pages/exploration-editor-page/services/change-list.service.ts#L80

Once I have made the new changes locally, this function will send the changelist back to the server, and `fetchLatestSavedData` will fetch this change list back (I checked the backend and function there is not checking the version id to save the change list, I am not sure whether this is a right way). 
https://github.com/oppia/oppia/blob/393a81474ddc02b7141eed80f33c720cde447517/core/controllers/editor.py#L689

Reply:
> Sorry, can you elaborate about which services specifically? I will further check. Thanks!

All the services which handle the editor pages like StateEditorServices, StateContentService, StateHintsService, etc. with the new data?

Note: I'm assuming that A (through the situation you have described in the PR explanation.) has done some changes in the exploration which will be fetched via `fetchLatestSavedData`.

Reply:
> So the current change is going to fetch the newest changes that A made as well as the change list B made, then apply the change list to the changes A made and then refresh the page.

Assume A changes the interaction for a card of exploration to "End exploration" (earlier it was "TextInput") whereas B has changelist which adds a hint to the same card. In this kinda scenerio I don't think we will be able to apply the changelist of B on top of new changes made by A.


I think the expected behavior for this issue isn't clear, @nithusha21 can you please explain or suggest what should be the expected behavior for this issue?

Reply:
I clarified the expected behaviour here: https://github.com/oppia/oppia/issues/4417#issuecomment-549573084. 

Does it help? Please let me know if you have further questions.